### DISCUSSION TITLE
Welcome to LibreDWG Discussions

### DISCUSSION THREAD

**Discussion by rurban** [LibreDWG/libredwg#301] [General]
## 👋 Welcome!
  We’re trying these Github Discussions as alternative to https://savannah.gnu.org/support/?group=libredwg. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 64sys** [comment]
Hello everyone, I am Martin Belmonte and I have been using CAD programs for 25 years and GNU / Linux for 15 years and Gambas for 6 years.
I belong to the GauchoCAD development team, where we are using libreDWG to open dwg files, there I take care of the GUI and the interaction with the files.
now I am reading and writing in json format, but I hope I can achieve the direct dialogue with the library and skip the intermediate step.
I thank you for the invitation to this group conversation.
Greetings to all.

**Comment by 64sys** [comment]
Hi Reini,
Actually I convert dwg to json (dwgread) then read that json and load in GauchoCAD making some gambas collections (JSONCOllections) for layers entities etc.
If I add some layer or for example a line, wich parameters i must to re arrange ? index? handle?
¿What means the four numbers in handle ?
Regards.

**Comment by rurban** [comment]
Martín <notifications@github.com> schrieb am Fr., 11. Dez. 2020, 19:57:

> Hi Reini,
> Actually I convert dwg to json (dwgread) then read that json and load in
> GauchoCAD making some gambas collections (JSONCOllections) for layers
> entities etc.
> If I add some layer or for example a line, wich parameters i must to re
> arrange ? index? handle?
>
A lot. See dwg_add_LAYER.
If you do it in json, to you want to write out that json to create a DWG?


¿What means the four numbers in handle ?
>
3 numbers. 1st is the code, 2nd the size, 3rd the value. Some codes are
relative, then the 4th number is the real value.


Regards.

**Comment by 64sys** [comment]
> A lot. See dwg_add_LAYER.
> If you do it in json, to you want to write out that json to create a DWG?
Actually I depend exclusively of json file made by dwgread, so I mod the info in Collections inside GauchoCAD then write a new json file then use:
**dwgwrite -y -I JSON -o tmp.dwg tmp.json**
I can do simple tasks like change color or coordinates but some more complex operations are out of my scope.
That is the reason i wan to mod the json file, but i know the better way is use directly the library, the objects rigth?
I want to use the library in GauchoCAD I'm trying to make a component outside gambas source code but not much success yet.
Regards.

**Comment by 64sys** [comment]
Hi,
I am trying to create the component in gambas in order to use libredwg in GauchoCAD
I'm learning so I'm going to go step by step.
What is the easiest data collection to obtain?
How does c_dwg.h work? Allows me to read and write directly to the dwg file? If so, to add a line or a layer, what information should I provide and to what object (class, collection, list, etc.)?
Regards

**Reply by rurban** [reply]
This belongs to https://github.com/LibreDWG/gambas3-bindings/discussions/1 really.
Note that the gambas bindings are not ready yet. We still need a proper component to use all the API's

**Reply by 64sys** [reply]
Yes, correct it belongs to LibreDWG/gambas3-bindings, I intent to made the component right now.
I know is not finish yet.

**Comment by 64sys** [comment]
Hi,
1) I already managed to create the component in gambas but it doesn't work at all.
What is the easiest thing I can do to interact with the library? query a list of blocks? layers? another thing?
2) I would like to ask you some conceptual questions about the library.
Assuming that in the near future we will be able to connect the library with Gambas, in that case:
Would it be possible to open a dwg in memory and modify existing entities, add new ones, or delete them and then save the dwg again?

**Comment by JoaoClementeSouza** [comment]
I would like to consult with the experts in this group.
I am trying to generate a copy of this library to run some tests in the hope of getting some information.

Let me describe my goal:
We have tons of DWGs files that reference other TIF files through the **XREF** feature.

I need to set up a process that is automatic and allows me to extract from within the DWG the complete path saved in the XREF.
My question is: _With this library, can I somehow get this "parameter"_ ? 

My hope was that this information would be exported somehow when converting the file to other more "**readable**" formats like JSON or even DWF...

If you have experience with this tool, could you tell me if it is possible to do that ?

Thanks

**Reply by rurban** [reply]
Sure. You can eg dwgread to export your files to json and check the xref paths there. Or write a C/C++ program to check it by yourself

**Comment by ReadXiao** [comment]
Hello author：
    My recent work needs to implement svg 2 dwg. Our svg diagram only has some simple line segments, etc., so I want to parse the line segment information in the svg diagram, and then write the line segment information into the dwg file through the libredwg library. I use the dwg_add_LINE function to add a line segment to Dwg_Data, but an error occurs when I use the dwg_wirte_file function to write Dwg_Data to the disk.
    Therefore, I would like to ask about the specific function of modifying the dwg file to disk. Is there a simple demo for this?
    I would be very grateful if you could reply to me!
